python/mcap/tests/test_reader.py

```diff
@@ -238,3 +238,17 @@ def test_no_summary_not_seeking(tmpdir: Path):
         assert len(list(NonSeekingReader(f).iter_attachments())) == 1
     with open(filepath, "rb") as f:
         assert len(list(NonSeekingReader(f).iter_metadata())) == 1
+
+
+def test_detect_invalid_initial_magic(tmpdir: Path):
+    filepath = tmpdir / "invalid_magic.mcap"
+    with open(filepath, "w") as f:
+        f.write("some bytes longer than the initial magic bytes")
+
+    with open(filepath, "rb") as f:
+        with pytest.raises(InvalidMagic):
+            SeekingReader(f)
+
+    with open(filepath, "rb") as f:
+        with pytest.raises(InvalidMagic):
+            NonSeekingReader(f).get_header()
```

Comment:
What if you use a NonSeekingReader but don’t explicitly ask for the header?

Reply:
The magic check only happens when [iterating over the records](https://github.com/foxglove/mcap/blob/8ac076a0bcd6a35ca7cdfc9dbddc345a2d5191d8/python/mcap/mcap/stream_reader.py#L104-L109) (which is what `get_header()` does). So just instantiating would not raise the invalid magic exception. Do we want to change this?

Reply:
I think that's fine, as long as it does happen sometime